src/lib/containers/entity_finder/tree/EntityTree.tsx

```diff
@@ -127,16 +130,7 @@ export function EntityTree(props: EntityTreeProps) {
     type: EntityDetailsListDataConfigurationType.PROMPT,
   }
 
-  const { accessToken } = useSynapseContext()
-
-  const [isLoading, setIsLoading] = useState(false)
-  const [topLevelEntities, setTopLevelEntities] = useState<
-    Pick<EntityFinderHeader, 'name' | 'id' | 'type'>[]
-  >([])
```

Comment:
These state variables were derived from other state variables and calculated in a `useEffect` block. Now, remove useEffect and derive this state in a `useMemo` hook below.